docs/plugin-development.rst

```diff
@@ -239,6 +239,37 @@ classes, and we can add it to the API in the ``load()`` method.
     def load(info):
         info['apiRoot'].cat = Cat()
 
+Adding a prefix to an API
+*************************
+
+It is possible to provide a prefix to your API, allowing associate endpoints to
+be grouped together. This is done by passing the resource instance
+when then route is defined and creating a prefix when mounting the resouce.
+
+
+.. code-block:: python
+
+    from girder.api.rest import Resource, Prefix
+
+    class Cat(Resource):
+        def __init__(self):
+            super(Cat, self).__init__()
+
+            self.route('GET', (), self.findCat, resource=self)
```

Comment:
I don't fully understand this yet -- how does passing `resource=self` actually change the behavior of the endpoint, relative to passing `resource=None`?

Reply:
If None is passed as resource then a default is set [here](https://github.com/girder/girder/blob/master/girder/api/rest.py#L784-L785), the default is a string rather than the resource instance. We need the resource instance so we can find out where it has been mounted. Does that make sense?

Reply:
I see. So making the default be `self` would *technically* be a breaking change, but practically, would it break anything?

Reply:
yes, probably not, in any case I think most plugins provide ```resourceName```, just being cautious.

Reply:
Aren't we past the 2.x-maintenance / master deviation? Can we start making breaking changes on master now?

Reply:
I was hoping you could answer that :smile:

Reply:
Part of me was hoping I could also get this a release relatively soon, so I could use it.

Reply:
That makes sense, but I don't want to propagate this usage that will be deprecated quite quickly in favor of omitting the argument... I'm tempted to change the default on this topic so that you can get this for free without having to pass an extra argument to every route. Am I correct in understanding that that would only break if you were already passing a `resource` parameter that was somehow different from where your resource was mounted?

Reply:
Correct. It is my understand that resource/resourceName is just used for the swagger doc, so currently we are relying on developer to mount things in the same place. So I think the only think we would break is people who have this mismatch in which can we are fixing them!?!?

Reply:
Right. So let's go ahead and make `self` the default value. This means we can also remove the `resourceName` attribute, right?

Reply:
@zachmullen and I spoke on hangout and we agreed to default to self ( this resource instance ) if ```resourceName``` is not provided and ```resource``` is ```None```. I will push this change.